common/hashing_test.cpp

```diff
@@ -20,54 +21,88 @@ using ::testing::Eq;
 using ::testing::Le;
 using ::testing::Ne;
 
-TEST(HashingTest, Integers) {
-  HashCode hash_zero = HashValue(0);
-  // Hashes should be stable within the execution.
-  EXPECT_THAT(HashValue(0), Eq(hash_zero));
-
-  for (int i : {0, 1, 2, 3, 42}) {
-    SCOPED_TRACE(llvm::formatv("Hashing: {0}", i).str());
-    HashCode hash = HashValue(i);
-
-    // Zero should match, and other integers shouldn't collide trivially.
-    if (i == 0) {
-      EXPECT_THAT(hash, Eq(hash_zero));
-    } else {
-      EXPECT_THAT(hash, Ne(hash_zero));
-    }
-
-    // Make sure all the different integer sizes hash. Note that while these
-    // *happen* to be equal for positive integers, there is no particular
-    // guarantee of that and negative integers have distinct hashes.
-    EXPECT_THAT(HashValue(static_cast<int8_t>(i)), Eq(hash));
-    EXPECT_THAT(HashValue(static_cast<uint8_t>(i)), Eq(hash));
-    EXPECT_THAT(HashValue(static_cast<int16_t>(i)), Eq(hash));
-    EXPECT_THAT(HashValue(static_cast<uint16_t>(i)), Eq(hash));
-    EXPECT_THAT(HashValue(static_cast<int32_t>(i)), Eq(hash));
-    EXPECT_THAT(HashValue(static_cast<uint32_t>(i)), Eq(hash));
-    EXPECT_THAT(HashValue(static_cast<int64_t>(i)), Eq(hash));
-    EXPECT_THAT(HashValue(static_cast<uint64_t>(i)), Eq(hash));
-  }
+TEST(HashingTest, HashCodeAPI) {
+  // Manually compute a few hash codes where we can exercise the underlying API.
+  HashCode empty = HashValue("");
+  HashCode a = HashValue("a");
+  HashCode b = HashValue("b");
+  ASSERT_THAT(HashValue(""), Eq(empty));
+  ASSERT_THAT(HashValue("a"), Eq(a));
+  ASSERT_THAT(HashValue("b"), Eq(b));
+  ASSERT_THAT(empty, Ne(a));
+  ASSERT_THAT(empty, Ne(b));
+  ASSERT_THAT(a, Ne(b));
+
+  EXPECT_THAT(HashValue("a").ExtractIndex(2), Eq(a.ExtractIndex(2)));
```

Comment:
Not sure what is being tested in this group of tests, is there any difference between `HashValue("a")` and `a`?